runtime/doc/lsp.txt

```diff
@@ -451,6 +451,84 @@ LspSignatureActiveParameter
+  LspType
+                                                                  *hl-LspClass*
+  LspClass
+                                                                   *hl-LspEnum*
+  LspEnum
+                                                              *hl-LspInterface*
+  LspInterface
+                                                                 *hl-LspStruct*
+  LspStruct
+                                                          *hl-LspTypeParameter*
+  LspTypeParameter
+                                                              *hl-LspParameter*
+  LspParameter
+                                                               *hl-LspVariable*
+  LspVariable
+                                                               *hl-LspProperty*
+  LspProperty
+                                                             *hl-LspEnumMember*
+  LspEnumMember
+                                                                  *hl-LspEvent*
+  LspEvent
+                                                               *hl-LspFunction*
+  LspFunction
+                                                                 *hl-LspMethod*
+  LspMethod
+                                                                  *hl-LspMacro*
+  LspMacro
+                                                                *hl-LspKeyword*
+  LspKeyword
+                                                               *hl-LspModifier*
+  LspModifier
+                                                                *hl-LspComment*
+  LspComment
+                                                                 *hl-LspString*
+  LspString
+                                                                 *hl-LspNumber*
+  LspNumber
+                                                                  *hl-LspRegex*
+  LspRegexp
+                                                               *hl-LspOperator*
+  LspOperator
+
+And the following highlight groups modifiers SemanticTokenModifiers:
+
+                                                               *hl-Declaration*
+  LspDeclaration
+                                                                *hl-Definition*
+  LspDefinition
+                                                                  *hl-Readonly*
+  LspReadonly
+                                                                    *hl-Static*
+  LspStatic
+                                                                *hl-Deprecated*
+  LspDeprecated
+                                                                  *hl-Abstract*
+  LspAbstract
+                                                                     *hl-Async*
+  LspAsync
+                                                              *hl-Modification*
+  LspModification
+                                                             *hl-Documentation*
+  LspDocumentation
+                                                            *hl-DefaultLibrary*
+  LspDefaultLibrary
+
```

Comment:
I did not document on how to support un-official modifiers like clangd's `scopeClass`, `scopeFile` as the implementation for more fine-grained mapping is subject to change. Also the tree-sitter style `cppLspComment` is undocumented as we need to agree on how to do more fine-grained mapping.

Reply:
Should I add a default mapping for all of these in this PR or in a follow-up?

Reply:
Personally, I'd leave them unmapped -- I have a hard time coming up with a good default that would not likely conflict with "base" highlighting. There's precedent: `LspReference*` are also unmapped by default. I'd add an explicit note in the docs that these are unmapped, though.

Reply:
This was also my thought when I created this PR. At the moment I have only mappings for `LspComment` and `LspDeprecated`

Reply:
That sounds reasonable -- then there's at least one "canary" group you can look for to see if it's working without further config.

Reply:
One could do the "obvious" mappings `LspFoo` -> `Foo` (if `Foo` is one of the builtin groups), but I fear that might be confusing (some groups are mapped and others aren't).